updatem/orchestration/update_orchestrator_feedback.go

```diff
@@ -0,0 +1,419 @@
+	if orchestrator.operation.status != types.StatusRunning {
+		return
+	}
+	domainStatus := orchestrator.operation.domains[domain]
+	if domainStatus != types.BaselineStatusActivationSuccess && domainStatus != types.BaselineStatusDownloadFailure &&
+		domainStatus != types.BaselineStatusUpdateFailure && domainStatus != types.BaselineStatusActivationFailure &&
+		domainStatus != types.StatusCompleted && domainStatus != types.StatusIncomplete {
+		return
+	}
+	orchestrator.operation.domains[domain] = types.BaselineStatusCleanupSuccess
+	orchestrator.domainUpdateCompleted(domain)
+}
+
+func (orchestrator *updateOrchestrator) handleDomainCleanupFailure(domain string, errMsg string) {
+	if orchestrator.operation.status != types.StatusRunning {
+		return
+	}
+	domainStatus := orchestrator.operation.domains[domain]
+	if domainStatus != types.BaselineStatusActivationSuccess && domainStatus != types.BaselineStatusDownloadFailure &&
+		domainStatus != types.BaselineStatusUpdateFailure && domainStatus != types.BaselineStatusActivationFailure &&
+		domainStatus != types.StatusCompleted && domainStatus != types.StatusIncomplete {
+		return
+	}
+	orchestrator.operation.domains[domain] = types.BaselineStatusCleanupFailure
+	orchestrator.domainUpdateCompleted(domain)
+}
+
+func (orchestrator *updateOrchestrator) handleDomainCleanup(domain string) {
+	if orchestrator.operation.status != types.StatusRunning {
+		return
+	}
+	domainStatus := orchestrator.operation.domains[domain]
+	if domainStatus != types.BaselineStatusActivationSuccess && domainStatus != types.BaselineStatusDownloadFailure &&
+		domainStatus != types.BaselineStatusUpdateFailure && domainStatus != types.BaselineStatusActivationFailure {
+		return
+	}
+	orchestrator.domainUpdateRunning()
+}
+
+func (orchestrator *updateOrchestrator) domainUpdateCompleted(domain string) {
+	for _, domainUpdateStatus := range orchestrator.operation.domains {
+		if domainUpdateStatus != types.BaselineStatusCleanupSuccess && domainUpdateStatus != types.BaselineStatusCleanupFailure {
+			return
+		}
+	}
+	for domain, actions := range orchestrator.operation.actions {
+		domainCfg := orchestrator.cfg.Agents[domain]
+		if domainCfg != nil {
+			if len(actions) > 0 && domainCfg.RebootRequired {
+				orchestrator.operation.rebootRequired = true
+				break
+			}
+		}
+	}
+	if orchestrator.operation.delayedStatus == types.StatusIncomplete {
+		orchestrator.operation.status = types.StatusIncomplete
+		orchestrator.operation.errMsg = "the update process is incompleted"
+		orchestrator.operation.errChan <- true
+		return
+	}
+	orchestrator.operation.status = types.StatusCompleted
+	orchestrator.operation.identDone <- true
+	orchestrator.operation.done <- true
+}
+
+func (orchestrator *updateOrchestrator) domainUpdateRunning() {
+	orchestrator.notifyFeedback(types.StatusRunning, "")
+}
+
+func (orchestrator *updateOrchestrator) updateStatusRunning() {
+	orchestrator.notifyFeedback(types.StatusRunning, "")
+}
```

Comment:
those functions are doing the same and are just one line wrapper, one of them is only used once, so can we get rid of them?

Reply:
Second function removed. The first one remains in order to follow similar pattern in all handler function.s